### DISCUSSION TITLE
Future of the project

### DISCUSSION THREAD

**Discussion by acaonweb** [stas-demydiuk/domoticz-zigbee2mqtt-plugin#849] [General]
Hi to everyone,
watching here about tho fantastic plug-in (and I can't thanks more and more Stanislav Demydiuk, I've noted that him isn't answering since February , which is the month when this shit war started.
Personally I'm worried
I think something happened, maybe someone have news?

**Comment by odelma** [comment]
I’m sure the situation in Ukraine has effect on how much time and effort Stas is able to put to this project. He has done commits during August/November and for other projects in October. Also there seems to be other people involved in this project who can merge and commit. So there is hopefully future for the project. 

But I do agree that to keep it alive and well some of the outstanding issues would need more attention. On the other hand, there is now the native autodiscovery for zigbee2mqtt which may have an effect on the motivation to keep this ”additional” plugin going. 

Personally I like some features of this plugin much more than in autodiscovery (like combined sensors, selectors for buttons etc) and therefore hope it will be kept alive 🙏

edit: especially I think it would be important to find some resolution to the "no more rows available" error (can be caused by Domoticz but anyhow, finding and fixing the reason is needed):
https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/issues/706
https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/discussions/817
https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/issues/791

...and then the generic "onMessage / typeError" related errors that seem to happen with specific devices, like the Aqara Opple. E.g. 
https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/issues/850
https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/issues/803
https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/issues/772

If these (not sure if they are related somehow, maybe both tied to specific devices as @stas-demydiuk does not see these in his environment) can be squashed, then I would see a brighter and clearer path to continue developing new features...👍

**Reply by stas-demydiuk** [reply]
@odelma, thanks for the feedback!

"no more rows available" is caused by a Domoticz upgrade, it never happens on a clean install but can happen if you upgrade a Domoticz. Not sure how we can fix/avoid it from the plugin side as the issue happens at Domoticz's DB layer

"onMessage / typeError" is usually related to error handling in the plugin itself when it receives the unexpected message from zigbee2mqtt, unfortunately as you've said it's hard to reproduce them without such devices, and as a result hard to fix, but I'm more than welcome if someone creates PR with the fix

**Reply by odelma** [reply]
Thanks for the reply, great to hear you are ok 🙏 

For "no more rows" I keep wondering why this plugin is the only one causing this issue? If by "clean install" you mean starting with empty db then that is something I'm afraid I cannot do. I was also thinking if there is something I could do for the database, some cleanup / maintenance routines perhaps. 

It does not help in troubleshooting that when reported in Domoticz (forum or github) the answer is usually to turn to plugin maintainer as they do not see this Domoticz error when this plugin is the only affected 😅 

Then on the other hand, these are just lines in the log and everything seems to work despite them. So I try not to worry too much about them 😄

**Reply by stas-demydiuk** [reply]
I guess the reason why the issue is rare and not reported by other plugins is because only a few plugins (or maybe this one even is the only one) that extensively use `DomoticzEx` plugin API instead of the older one `Domoticz` API. And this plugin uses `DomoticzEx` because it allows to overcome 255 devices limit

**Reply by odelma** [reply]
Indeed, this could be the case that there is not too many plugins using the DomoticzEx at the moment. Well, as it does not seem to cause any adverse effects, I will just continue to dismiss these errors. And the generic onMessage / typeError seems to have been largely fixed by https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/commit/389b422e949581fac5b9f00a85aa1c9703ac016a - big thanks! 😃

**Comment by stas-demydiuk** [comment]
@acaonweb thank you for asking, I'm alright and I still support the plugin. The development is not as active as it was in the past as the plugin already supports all most-common devices and for very-specific ones, I'd prefer the contribution from the community. But if many people ask for the same feature I can take a look too.

**Reply by acaonweb** [reply]
I'm glad to hear news from you. 
I've two kind of PIR, one is aqara and works fine. 
One is a different hardware and make crash plugin. 
Let me know what kind of output from zigbee2mqtt you need. 
But it's important to fix that because all the plugin stops to work and the only way to use the hardware that is ok for zigbe2mqtt is switch to domoticz auto discover that is far far away from this great plugin.
And I know some people that is leaving it. and it's really pity. 
Please consider to have a look to my open issue about it,
Thanx so much. 

Fabrizio.

**Reply by acaonweb** [reply]
https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/issues/847
